### ISSUE TITLE
Dropout with multiple layer network

### ISSUE THREAD

**Issue by MordFee** [fchollet/keras#1716]
Hello,

I have tried the dropout feature with multiple layers, and get a problem I couldn't figure out.

With Dropout = [0.8, 0.5, 0.5] for input layer and the 2 hidden layers,
and the structure Layers = [784, 300, 100, 10] with MNIST database, I get this error:

> File X in fit_network
    output_file=output_filepath
  File "X\keras\models.py", line 438, in fit
    shuffle=shuffle, metrics=metrics, output_file=output_file)
  File "X\keras\models.py", line 192, in \_fit
    outs = f(\*ins_batch)
  File "X\Anaconda\lib\site-packages\theano\compile\function_module.py", line 606, in _ _call__
    storage_map=self.fn.storage_map)
  File "X\Anaconda\lib\site-packages\theano\compile\function_module.py", line 595, in _ _call__
    outputs = self.fn()

> ValueError:
 Input dimension mis-match. (input[0].shape[0] = 100, input[4].shape[0] = 300)
Apply node that caused the error: Elemwise{Composite{(((i0 + (i1 * i2)) - i3) * i4)}}[(0, 0)](<TensorType(float64, matrix)>, TensorConstant{(1L, 1L) of 0.9}, Elemwise{Composite{((i0 * i1) - i2)}}[(0, 1)].0, Dot22Scalar.0, TensorConstant{[[ 1.  1. ..  0.  1.]]})
Inputs types: [TensorType(float64, matrix), TensorType(float64, (True, True)), TensorType(float64, matrix), TensorType(float64, matrix), TensorType(float64, matrix)]
Inputs shapes: [(100L, 10L), (1L, 1L), (100L, 10L), (100L, 10L), (300L, 100L)]
Inputs strides: [(80L, 8L), (8L, 8L), (80L, 8L), (80L, 8L), (800L, 8L)]
Inputs values: ['not shown', array([[ 0.9]]), 'not shown', 'not shown', 'not shown']

> HINT: Re-running with most Theano optimization disabled could give you a back-trace of when this node was created. This can be done with by setting the Theano flag 'optimizer=fast_compile'. If that does not work, Theano optimizations can be disabled with 'optimizer=None'.
HINT: Use the Theano flag 'exception_verbosity=high' for a debugprint and storage map footprint of this apply node.

**Comment by MordFee** [comment]
BTW,
I have tried the same network without dropout -> it works fine
and a 1 hidden layer network with dropout -> it works fine.

I don't understand why it doesn't work when I get the two together

**Comment by pasky** [comment]
Can you include the code you use to build the model?

**Comment by MordFee** [comment]
ok, nevermind
this problem came from code I modified in keras.
The official version works just fine.
